src/MongoDB.Driver.Core/Core/Connections/BinaryConnection.cs

```
@@ -727,7 +733,8 @@ private Exception WrapException(Exception ex, string action)
                 ex is StackOverflowException ||
 #endif
                 ex is MongoAuthenticationException ||
-                ex is OutOfMemoryException)
+                ex is OutOfMemoryException ||
+                ex is OperationCanceledException)
```

we should throw the cancellation exception to the higher levels